### PR TITLE
android, cronet, android-interop-testing: bump Android versions

### LINKED ISSUE #7439: Upgrade compile/target Android SDK version to 29
`grpc-cronet` (currently SDK 27) should be easy.

`grpc-android` (currently SDK 28) uses `android.net.NetworkInfo` to detect network connectivity for API < 24, which has been [deprecated](https://developer.android.com/reference/android/net/NetworkInfo) in SDK 29. This may block us upgrading the compile SDK version.

`grpc-android-interop-testing` (currently SDK 26) needs some cleanup for errorprone and need to [migrate android support library to AndroidX](https://developer.android.com/jetpack/androidx/migrate).

### LINKED ISSUE #7734: Fix lint issues in Android projects
It seems we aren't targeting the a recent Android version. Many of the CheckResult failures appear to be expected behavior, so it will need to be silenced in some way. There's some warnings about old versions and some UI goo. Quite a mixed bag. I'm fine if we resolve a substantial portion of them with this issue and open separate issues for the remaining items.

```
> Task :grpc-android:lint
Ran lint on variant debug: 1 issues found
Ran lint on variant release: 1 issues found
Wrote HTML report to grpc-java/android/build/reports/lint-results.html
Wrote XML report to grpc-java/android/build/reports/lint-results.xml

> Task :grpc-cronet:lint
Ran lint on variant debug: 1 issues found
Ran lint on variant release: 1 issues found
Wrote HTML report to grpc-java/cronet/build/reports/lint-results.html
Wrote XML report to grpc-java/cronet/build/reports/lint-results.xml

> Task :grpc-android-interop-testing:lint
Ran lint on variant release: 64 issues found
Ran lint on variant debug: 64 issues found
Wrote HTML report to grpc-java/android-interop-testing/build/reports/lint-results.html
Wrote XML report to grpc-java/android-interop-testing/build/reports/lint-results.xml
```

### PR DESCRIPTION
**Pull request by voidzcy**
Bump target SDK version of grpc-android, grpc-cronet, grpc-android-interop-testing to API 29. Resolves #7439.
- Migrated `android.support` to `androidx`, which is required for Android Q.
- `android.net.NetworkInfo` is deprecated in SDK 29, suppressed compiler warnings as it is needed for supporting old Android versions.
- Robolectric requires Java 9+ for Android 29, which causes unit tests in grpc-cronet fail. Added annotation to emulate Android 28 for grpc-cronet's test.
- Upgraded android dependencies accordingly in android-interop-testing.

This is also a partial fix for #7734 (grpc-android and grpc-cronet should be clean now, android-interop-testing still has some code lint issues need to be fixed). Also, android examples are not updated.

### CODE REVIEW

File: android-interop-testing/build.gradle [hunk 1]
```diff
@@ -47,10 +47,10 @@ android {
 }
 
 dependencies {
-    implementation 'com.android.support:appcompat-v7:26.1.0'
-    implementation 'com.android.support:multidex:1.0.3'
-    implementation 'com.android.support:support-annotations:27.1.1'
-    implementation 'com.google.android.gms:play-services-base:15.0.1'
+    implementation 'androidx.appcompat:appcompat:1.3.0-alpha02'
```
Comment (ejona86) [inline android-interop-testing/build.gradle:50]: Why use alpha versions?

Comment (voidzcy) [reply android-interop-testing/build.gradle:50]: Shall we avoid using alpha versions? I was just upgrade everything to versions as latest as possible.

Comment (voidzcy) [reply android-interop-testing/build.gradle]: Changed to use stable versions.

**Comment by voidzcy** [comment]
Note after this change, you may still see tons of 
```
warning: [MissingSummary] A summary line is required on public/protected Javadocs.
    (see http://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment)
```
in android-interop-testing. Those are hard to fix. The message doesn't give any clue for where the source code is. Will fix it separately.

File: android-interop-testing/build.gradle (hunk 1 above)
Comment (ejona86) [reply android-interop-testing/build.gradle]: Yeah, we prefer stable versions.

**Comment by ejona86** [comment]
> Note after this change, you may still see tons of... MissingSummary...

Yeah, I had seen that before. I think https://github.com/google/error-prone/issues/1981 is related, although not the exact same trigger. I think I tracked it down to the gRPC generated code, but couldn't figure out why our filters weren't excluding it from checking.

**Review by ejona86 — APPROVED** [review]

**Pull request merged**